proto/src/determined/api/v1/api.proto

```diff
@@ -167,7 +167,16 @@ service Determined {
         option (google.api.http) = {post: "/api/v1/experiments/{id}/unarchive" };
         option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Experiments"};
     }
-
+    // Set labels for an experiment.
+    rpc SetExperimentLabels(SetExperimentLabelsRequest) returns (SetExperimentLabelsResponse) {
+        option (google.api.http) = {patch: "/api/v1/experiments/{id}/labels" body: "*" };
+        option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Experiments"};
+    }
+    // Set description for an experiment.
+    rpc SetExperimentDescription(SetExperimentDescriptionRequest) returns (SetExperimentDescriptionResponse) {
+        option (google.api.http) = {patch: "/api/v1/experiments/{id}/description" body: "*" };
+        option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Experiments"};
+    }
```

Comment:
Coming back to this one again, but do you think this should be a `POST`, `PATCH` or `PUT`. I can see arguments for all - usually I just saw it's an RPC to update a subfield of a resource and that's a `POST`, but I could also see it being a total update of `/description` or `/labels` and a `PUT`.

In other places, I think requests like this have an `POST`s. I'm ok with changing that, but we should go ahead and make a solid decision and stick to it.